e2e-test/src/main.rs

```diff
@@ -124,21 +125,26 @@ async fn main() {
 
     let test_cfg = TestCfg::init().expect("Failed configuration initialization");
 
-    let mut sdk_options = AstarteOptions::new(
+    let mut mqtt_config = MqttConfig::new(
         &test_cfg.realm,
         &test_cfg.device_id,
         &test_cfg.credentials_secret,
         &test_cfg.pairing_url,
-    )
-    .interface_directory(&test_cfg.interfaces_fld.to_string_lossy())
-    .unwrap();
+    );
 
     // Ignore SSL for local testing
     if env::var("E2E_IGNORE_SSL").is_ok() {
-        sdk_options = sdk_options.ignore_ssl_errors();
+        mqtt_config.ignore_ssl_errors();
     }
 
-    let (mut device, mut rx_events) = AstarteDeviceSdk::new(sdk_options).await.unwrap();
+    let (mut device, mut rx_events) = DeviceBuilder::new()
+        .store(MemoryStore::new())
+        .interface_directory(&test_cfg.interfaces_fld.to_string_lossy())
+        .unwrap()
```

Comment:
I would use `expect` rather then `unwrap` since you can explain the error

Reply:
For now this PR just adapted the use of the new builder the e2e test is full of `unwraps`s but this will be fixed in a future pr. I've also opened a [tracking issue](https://github.com/astarte-platform/astarte-device-sdk-rust/issues/256).